es6/experiment.js

```diff
@@ -7,27 +7,17 @@ export default function provideExperiment(Assignment) {
       this._exposureLogged = false;
       this._salt = null;
       this._inExperiment = true;
-
-      this.name = this.getDefaultExperimentName();
```

Comment:
I think it naturally works with classes and it isn't clear to me what other designs would result in things being cleaner. 

Do you have some ideas for a more functional implementation?